### PR TITLE
A much smaller set of newly enabled linters

### PR DESCRIPTION
**Pull request by faddat**
## Description

This PR enables many more linters, but not nearly as many as the one that it is replacing.

The reason it hits so many files is this:

https://github.com/mvdan/gofumpt

From here, we can re-enable additional linters in .golangci.yml as needed.

______

For contributor use:

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -4,23 +4,98 @@ run:
 #   timeout: 5m
 
 linters:
+  # Disable all linters.
+  # Default: false
```
Comment (alexanderbez) [inline .golangci.yml:8]: nit: I would remove any commented out sections unless we intend to re-enable them in the future.

```suggestion
```

File: app/upgrades/v5/whitelist_feetokens.go [hunk 2]
```diff
@@ -13,6 +13,7 @@ import (
 // Every asset with a liquid osmo pairing pool on Osmosis, as of 12/01/21
 // Notably, Tick is not on this list because the osmo pool has $76 of liquidity.
 // Cheq'd and KRT are also not on this, due to neither having osmo pairings.
+//nolint:gosec // nolint because these are strings of whitelisted ibc denoms
```
Comment (alexanderbez) [inline app/upgrades/v5/whitelist_feetokens.go:16]:
```suggestion
// We nolint because these are strings of whitelisted ibc denoms.
//nolint:gosec
```

**Review by alexanderbez — APPROVED** [review]
Skimmed through a bunch of the changes and it LGTM. I left one minor comment on the config file.

Also, this might clobber a bit with the removed annotations in the import blocks in my PR #1125, but I can resolve conflicts if need be.

**Review by ValarDragon — APPROVED** [review]
LGTM as well! Thanks for pairing this down from the prior PR. RIP I made merge conflicts with merging #1129 , I'll fix the conflicts here

**Comment by ValarDragon** [comment]
Actually, if its ok with you all, I'd really like to disable godot. (We can keep the changes)

`godot` just feels like an unnecessary headache. Also we need to update `make lint` to have parity with this

**Comment by ValarDragon** [comment]
Latest commit:
* Updates `make lint` to use the config file
* removes godot (indefinitely if noone has other thoughts)
* removes gofumpt (only until we get make lint auto-format integration sorted & vscode / makefile installation worked out)

**Comment by alexanderbez** [comment]
Sounds good to me @ValarDragon

**Comment by faddat** [comment]
this all sounds good to me.   Just updating changelog now.

**Pull request merged**